### PR TITLE
Use moby/sys/mountinfo

### PR DESCRIPTION
**Pull request by kolyshkin**
*(this is a followup to https://github.com/rootless-containers/rootlesskit/pull/122)*

Drop the implementation forked from runc, use moby/sys/mountinfo.

Use SingleEntryFilter as we're only interested in / mount.

NOTE we're no longer checking for m.Root == "/", only for
the mount point.

### CODE REVIEW

**Comment by kolyshkin** [comment]
@AkihiroSuda @thaJeztah PTAL

**Comment by kolyshkin** [comment]
Can't figure out why CI fails :(

File: pkg/parent/warn.go [hunk 1]
```diff
@@ -5,27 +5,17 @@ import (
...
-	for _, m := range mounts {
-		if m.Root == "/" && m.Mountpoint == "/" {
-			root = m
-		}
-	}
-	if root == nil {
-		logrus.Warn("Failed to parse mountinfo of the root filesystem")
-		return
-	}
+	root = mounts[0]
```
Comment (AkihiroSuda) [inline pkg/parent/warn.go]:
```
# github.com/rootless-containers/rootlesskit/pkg/parent
src/github.com/rootless-containers/rootlesskit/pkg/parent/warn.go:18:2: undefined: root
src/github.com/rootless-containers/rootlesskit/pkg/parent/warn.go:23:68: undefined: root
src/github.com/rootless-containers/rootlesskit/pkg/parent/warn.go:25:4: undefined: root
```

**Comment by AkihiroSuda** [comment]
Thanks, LGTM except the compilation failure

**Comment by kolyshkin** [comment]
oops, fixed

**Pull request merged**